src/biogeophys/BalanceCheckMod.F90

```diff
@@ -159,10 +204,98 @@ subroutine BeginWaterBalance(bounds, &
+    integer :: begc, endc, begg, endg  ! bounds
+    real(r8) :: qflx_liq_dynbal_left_to_dribble(bounds%begg:bounds%endg)  ! grc liq dynamic land cover change conversion runoff flux at beginning of time step
+    real(r8) :: qflx_ice_dynbal_left_to_dribble(bounds%begg:bounds%endg)  ! grc ice dynamic land cover change conversion runoff flux at beginning of time step
+    !-----------------------------------------------------------------------
+
+    associate(                                     &
+         zi        => col%zi                     , &  ! Input:  [real(r8) (:,:) ]  interface level below a "z" level (m)
+         zwt       => soilhydrology_inst%zwt_col , &  ! Input:  [real(r8) (:)   ]  water table depth (m)
+         aquifer_water_baseline => waterstate_inst%aquifer_water_baseline, &  ! Input: [real(r8)] baseline value for water in the unconfined aquifer (wa_col) for this bulk / tracer (mm)
+         wa        => waterstate_inst%wa_col     , &  ! Output: [real(r8) (:)   ]  water in the unconfined aquifer (mm)
+         begwb_col => waterbalance_inst%begwb_col, &  ! Output: [real(r8) (:)   ]  column-level water mass begining of the time step
+         begwb_grc => waterbalance_inst%begwb_grc  &  ! Output: [real(r8) (:)   ]  grid cell-level water mass begining of the time step
+         )
+
+    begc = bounds%begc
+    endc = bounds%endc
+    begg = bounds%begg
+    endg = bounds%endg
+
+    ! wa(c) gets added to liquid_mass in ComputeLiqIceMassNonLake
+    if(use_aquifer_layer) then
+       do fc = 1, num_nolakec
+          c = filter_nolakec(fc)
+          if (col%hydrologically_active(c)) then
+             if(zwt(c) <= zi(c,nlevsoi)) then
+                wa(c) = aquifer_water_baseline
+             end if
+          end if
+       end do
+    endif
+
+    ! NOTES subroutines Compute*Mass* are in TotalWaterAndHeatMod.F90
+    !       endwb is calculated in HydrologyDrainageMod & LakeHydrologyMod
+    call ComputeWaterMassNonLake(bounds, num_nolakec, filter_nolakec, &
+         waterstate_inst, waterdiagnostic_inst, &
+         subtract_dynbal_baselines = .false., &
+         water_mass = begwb_col(begc:endc))
+
+    call ComputeWaterMassLake(bounds, num_lakec, filter_lakec, &
+         waterstate_inst, lakestate_inst, &
+         add_lake_water_and_subtract_dynbal_baselines = .false., &
+         water_mass = begwb_col(begc:endc))
+
+    call c2g(bounds, begwb_col(begc:endc), begwb_grc(begg:endg), &
+             c2l_scale_type='urbanf', l2g_scale_type='unity')
+
+    call waterflux_inst%qflx_liq_dynbal_dribbler%get_amount_left_to_dribble_beg(bounds, qflx_liq_dynbal_left_to_dribble(begg:endg))
+    call waterflux_inst%qflx_ice_dynbal_dribbler%get_amount_left_to_dribble_beg(bounds, qflx_ice_dynbal_left_to_dribble(begg:endg))
+
+    do g = begg, endg
+       begwb_grc(g) = begwb_grc(g) - qflx_liq_dynbal_left_to_dribble(g)  &
+                                   - qflx_ice_dynbal_left_to_dribble(g)
+    end do
```

Comment:
I struggled with this during testing. I had a hard time convincing myself that the `+` sign was wrong. But as soon as I decided to flip the sign, the previously failing conservation test passed. Two hints led me to flip the sign:
- The size of the conservation error
- The `-` signs here:
https://github.com/ESCOMP/CTSM/blob/097ff9805c5d10e55c9be42e6ec19d066a3b963f/src/main/lnd2atmMod.F90#L376-L379
https://github.com/ESCOMP/CTSM/blob/097ff9805c5d10e55c9be42e6ec19d066a3b963f/src/main/lnd2atmMod.F90#L396-L398

I would not mind a suggestion regarding the wording, since you offered.

Reply:
How about something like this (I don't love it, but it's something...):

These dynbal dribblers actually store the delta state, (end - beg). Thus, the amount dribbled out is the negative of the amount stored in the dribblers. Similarly, to achieve conservation, we need to subtract (rather than add) the amount remaining to dribble.

Reply:
Thank you @billsacks 
I slightly reworded the comment and added to both locations.

I will not investigate this further unless you want me to; however, I can't help but wonder why the C-balance code uses `+` signs in CNBalanceCheck. Without digging deeper, I suspect an inconsistency in the handling of the C vs. H2O dribble terms. Let me know if you'd like me to look into it.

Reply:
I think you're right that the two are inconsistent, and that's my fault. In retrospect, I think the convention I used for the water and energy dribblers is counter-intuitive. Probably what we should do is to store the negative of what we currently store in those dribblers. To be honest, I'm not sure I have the mental energy to make this change right now, but I'm certainly open to your doing it if you think the benefit would be significant. One of the things stopping me from doing it is that it will cause incorrect fluxes in the first year of any transient run that starts mid-year from an old restart file (with the current sign convention). That probably won't impact many people (and the impact should be small), so that is *probably* okay if you are inclined to tackle this. I'm fine either way (leaving it for now or your spending the time to change this).

Reply:
I'm fine not addressing it now. If you don't mind, I can include some of this info in the comment that I'm adding about the `-` sign.